pkg/segment/structs/metricsstructs.go

```diff
@@ -141,10 +143,35 @@ type QueryArithmetic struct {
 	Operation   utils.LogicalAndArithmeticOperator
 	Constant    float64
 	// maps groupid to a map of ts to value. This aggregates DsResults based on the aggregation function
-	Results       map[string]map[uint32]float64
-	OperatedState bool //true if operation has been executed
+	Results        map[string]map[uint32]float64
+	OperatedState  bool //true if operation has been executed
+	VectorMatching *VectorMatching
 }
 
+// VectorMatching describes how elements from two Vectors in a binary
+// operation are supposed to be matched.
+type VectorMatching struct {
+	// The cardinality of the two Vectors.
+	Card VectorMatchCardinality
```

Comment:
nit: rename `Card` to `Cardinality`; this makes the meaning much more clear, for only 7 extra characters